krod/core/vector_store.py

```
@@ -49,45 +84,105 @@ def add_document(self, text: str, metadata: Dict[str, Any] = None) -> str:
+                points=points,
+                wait=True
+            )
+            
+            self.logger.info(f"Added {len(texts)} documents to collection {self.collection_name}")
+            return ids
+            
         except Exception as e:
-            self.logger.error(f"Failed to add document: {str(e)}")
-            raise RuntimeError(f"Failed to add document: {str(e)}")
+            self.logger.error(f"Failed to add documents: {str(e)}")
+            raise RuntimeError(f"Failed to add documents: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)